src/formatDate.js

```diff
@@ -49,8 +49,68 @@
  * @returns {string}
  */
 
+function getNewFormatYear(fromFormYear, toFormYear, oldFormatYear) {
+  let newFormatYear = oldFormatYear;
```

Comment:
It is a little unclear why you assign a variable of the new format to a variable of the old format.
It will be enough to `return` the changed value for each `if`, then there will be no need to create an extra variable.